map_model/src/objects/traffic_signals.rs

```diff
@@ -71,6 +79,27 @@ impl ControlTrafficSignal {
         brute_force(map, id)
     }
 
+    pub fn get_min_crossing_time(&self, idx: usize) -> Duration {
+        let mut max_distance = Distance::meters(0.0);
+        for movement in &self.stages[idx].protected_movements {
+            if movement.crosswalk {
+                let distance = self.movements.get(&movement).unwrap().geom.length();
+                if max_distance == Distance::meters(0.0) {
+                    max_distance = distance;
+                } else if max_distance < distance {
+                    max_distance = distance;
+                }
+            }
+        }
+        let time = max_distance / CROSSWALK_PACE;
+        if time >= Duration::ZERO {
+            //Round up because it is converted to a usize elsewhere
+            Duration::seconds(time.inner_seconds().ceil())
+        } else {
+            Duration::seconds(5.0)
```

Comment:
This case is impossible, distance of turns is never negative. You could put an `assert!(time >= Duration::ZERO);` a few lines up instead

Reply:
Yup, good spot. Thanks!